### DISCUSSION TITLE
modelling small peptides with D-amino acids

### DISCUSSION THREAD

**Discussion by imerelli** [RosettaCommons/rosetta#178] [Rosetta]
Hello, I need to model many small peptides with D-amino acids (passing only the sequence because I need to model a large number of them).

For L-amino acids, I simply use:

relax.static.linuxgccrelease -in:file:fasta peptide.fa -relax:fast -out:path:all ./output

But I just can’t figure out how to do this for D-amino acids. I don’t understand whether I need to use -packing:resfile by passing some instructions indicating that the amino acids are D-form (although I wouldn’t know how to do this) or if I need to pass .params files for the D-amino acids, which I would have created anyway.

Can you help me?

**Comment by roccomoretti** [comment]
Unfortunately, the `-in:file:fasta` commandline option assumes that you have a standard (canonical) amino acid sequence.

However, there are ways to use the "annotated sequence" format for more control over which residues get instantiated. However, it doesn't look like this is exposed to the command line yet, though you can access it from within RosettaScripts with the [PoseFromSequenceMover](https://docs.rosettacommons.org/docs/latest/scripting_documentation/RosettaScripts/Movers/movers_pages/PoseFromSequenceMover).

A simple script like the following should work:
```
<ROSETTASCRIPTS>
	<MOVERS>
                <PoseFromSequenceMover name="build_pose" fasta="%%fasta_in%%" />
                <FastRelax name="relax" />
	</MOVERS>
	<PROTOCOLS>
                <Add mover="build_pose" />
                <Add mover="relax" />  -- Optional. If you do want to relax the peptide.
	</PROTOCOLS>
</ROSETTASCRIPTS>
```

Then you can call it on the command line like `rosetta_scripts.linuxgccrelease -input_empty_pose true -parser:protocol build_pose_script.xml -parser:script_vars fasta_in=peptide.fa -out:path:all ./output`

Then in your annotated sequence file you simply need to provide the full name (not the three letter code) of the non-canonical residues you wish to include.

For example, to build a peptide with alternating L and D amino acids:
```
>sample.fasta
AA[DALA]CC[DCYS]DD[DASP]EE[DGLU]FF[DPHE]GHH[DHIS]II[DILE]KK[DLYS]LL[DLEU]MM[DMET]NN[DASN]PP[DPRO]QQ[DGLN]RR[DARG]SS[DSER]TT[DTHR]VV[DVAL]WW[DTRP]YY[DTYR]
```

Note that you shouldn't need to build params files for the D-amino acids -- there should be parameters for them already in the database. You would only need to bother to make params files for non-canonical which Rosetta doesn't already know about. But using those non-canonicals is similar to using the D amino acids -- just provide the full `NAME` of the residue after the corresponding one-letter `IO_STRING` code from the params file. .. And if you want to use D-variants of existing L-NCAAs, it might just be as simple as prefixing a `D` to the front of the name. (D amino acid types are handled as "patches" to the L-types. That patch can potentially be applied to other NCAA L amino acids as well.)

The one downside of this approach is that the output filenames are listed as "S_0001.pdb", rather than named according to the name of the input fasta. You can try to compensate for this by adding something like `-out:file:prefix sample_` to the command line, where you manually set the prefix for the peptide you want to build.

**Comment by vmullig** [comment]
See the documentation on packer pallettes and resfiles. It should be
possible to do this from the command line now.

**Reply by roccomoretti** [reply]
Note that packer palettes and resfiles are used when designing to D-amino acids. They're not necessarily involved with the initial reading-in of structures.

**Comment by vmullig** [comment]
Ah, sorry.  I thought you were designing with D-amino acids.
